exercises/allergies/src/allergies.h

```diff
@@ -10,12 +12,12 @@ typedef enum {
    ALLERGEN_CHOCOLATE,
    ALLERGEN_POLLEN,
    ALLERGEN_CATS,
-   ALLERGEN_COUNT
+   ALLERGEN_COUNT,
 } allergen_t;
 
 typedef struct {
    int count;
```

Comment:
Does it make sense to include `count` now? The size of `allergens` is fixed and the allergens in the list are not sorted by their presence, so it's difficult to see how `count` is adding any value.

Reply:
Oh, I thought I explained this in my PR comment but apparently not.

I left `count` because I noticed that the test suite as it stands has the issue that it never does a negative test while returning the list. So just returning an array of all `true` would work for every test.

This suite is mostly equivalent to the old version, which was my original goal with the changes, but still has that rather glaring issue in it. To be honest though, I mostly stopped at this because I was lazy and didn't feel like finishing the full overhaul of the tests.

It might be worth it to go ahead and update to the new test suite while improving this. I would remove this `count` and add back a helper function that takes a list of allergens and validates that only those are set in the array, and then update all the tests to match. The error messages will be a _lot_ worse for that though.

Reply:
Got it. Let's go ahead and merge this then we'll address the remaining issues when we overhaul the test suite to get it back in sync.